### DISCUSSION TITLE
2024 Roadmap

### DISCUSSION THREAD

**Discussion by akshayjshah** [connectrpc/.github#16] [Announcements]
# Roadmap for 2024

Since [announcing Connect][announcement] in the summer of 2022, we've shipped implementations in [Go][connect-go], [TypeScript][connect-es], [Swift][connect-swift], and [Kotlin][connect-kotlin]. Our heartfelt thanks go out to everyone using Connect, submitting bug reports and pull requests, and chatting with us in [Slack](https://buf.build/links/slack) &mdash; we couldn't have come this far without you!

As always, most of the work we're planning to do in 2024 is tracked in Github issues in the individual repositories. However, we are planning to work on a few larger initiatives.

## Language-Agnostic

* Join the Cloud Native Computing Foundation ([application][cncf-application]).
* Expand and stabilize Connect's [conformance tests][conformance].
* Begin at least one new Connect implementation (likely Python, Rust, Java, or C++).

## Go

The Go implementation is fairly mature, so most of our effort will focus on conformance testing and quality-of-life improvements.

* Stable [OpenTelemetry tracing and metrics][otelconnect].
* Stable [authentication middleware][authn-go].
* Stable [CORS middleware][cors-go].

## TypeScript

In TypeScript, we'll focus on additional JavaScript runtimes and preparing for the next evolution of the Protobuf schema language.

* Thorough testing of the Cloudflare Workers and Vercel serverless runtimes.
* Though not strictly part of the Connect project, many Connect committers are getting [protobuf-es] ready for the upcoming release of [Protobuf Editions][editions].
* Time permitting, additional quality-of-life improvements.

## Mobile

* Stable release of the [Swift runtime][connect-swift].
* Stable release of the [Kotlin runtime][connect-kotlin].

[matt]: https://github.com/mattrobenolt
[eddie]: https://github.com/eseay
[cncf-application]: https://github.com/cncf/sandbox/issues/63
[connect-go]: https://github.com/connectrpc/connect-go
[connect-es]: https://github.com/connectrpc/connect-es
[connect-swift]: https://github.com/connectrpc/connect-swift
[connect-kotlin]: https://github.com/connectrpc/connect-kotlin
[announcement]: https://buf.build/blog/connect-a-better-grpc
[otelconnect]: https://github.com/connectrpc/otelconnect-go
[editions]: https://protobuf.dev/editions/overview/
[protobuf-es]: https://github.com/bufbuild/protobuf-es
[conformance]: https://github.com/connectrpc/conformance
[authn-go]: https://github.com/connectrpc/authn-go
[cors-go]: https://github.com/connectrpc/cors-go

**Comment by Binlogo** [comment]
Expect for Connect implementation in Rust 👀

**Comment by jmzwcn** [comment]
Great work, thank you~

**Comment by caseylucas** [comment]
Too early to inquire about a 2025 roadmap 😉? I'm particularly interested in c#/.net support.

**Reply by tjb1982** [reply]
Definitely not too early for a 2025 roadmap now. Interested in Rust personally.

**Reply by spacebat** [reply]
Elixir/Erlang would be nice

**Comment by andres-ito1** [comment]
Is there a 2025 or 2026 roadmap available?
